src/main/java/com/yandex/money/api/typeadapters/ProcessPaymentTypeAdapter.java

```diff
@@ -0,0 +1,125 @@
+    private static final String MEMBER_HOLD_FOR_PICKUP_LINK = "hold_for_pickup_link";
+    private static final String MEMBER_PAYEE = "payee";
+    private static final String MEMBER_PAYEE_UID = "payee_uid";
+    private static final String MEMBER_PAYER = "payer";
+    private static final String MEMBER_PAYMENT_ID = "payment_id";
+
+    private ProcessPaymentTypeAdapter() {
+    }
+
+    /**
+     * @return instance of this class
+     */
+    public static ProcessPaymentTypeAdapter getInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public ProcessPayment deserialize(JsonElement json, Type typeOfT,
+                                      JsonDeserializationContext context)
+            throws JsonParseException {
+
+        JsonObject object = json.getAsJsonObject();
+
+        ProcessPayment.Builder builder = new ProcessPayment.Builder()
+                .setPaymentId(getString(object, MEMBER_PAYMENT_ID))
+                .setBalance(getBigDecimal(object, MEMBER_BALANCE))
+                .setPayer(getString(object, MEMBER_PAYER))
+                .setPayee(getString(object, MEMBER_PAYEE))
+                .setCreditAmount(getBigDecimal(object, MEMBER_CREDIT_AMOUNT))
+                .setAccountUnblockUri(getString(object, MEMBER_ACCOUNT_UNBLOCK_URI))
+                .setPayeeUid(getString(object, MEMBER_PAYEE_UID))
+                .setHoldForPickupLink(getString(object, MEMBER_HOLD_FOR_PICKUP_LINK))
+                .setDigitalGoods(deserializeDigitalGoods(object));
+        BaseProcessPaymentTypeAdapter.Delegate.deserialize(object, builder);
+        return builder.create();
+    }
+
+    @Override
+    public JsonElement serialize(ProcessPayment src, Type typeOfSrc,
+                                 JsonSerializationContext context) {
+
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(MEMBER_PAYMENT_ID, src.paymentId);
+        jsonObject.addProperty(MEMBER_BALANCE, src.balance);
+        jsonObject.addProperty(MEMBER_PAYER, src.payer);
+        jsonObject.addProperty(MEMBER_PAYEE, src.payee);
+        jsonObject.addProperty(MEMBER_CREDIT_AMOUNT, src.creditAmount);
+        jsonObject.addProperty(MEMBER_ACCOUNT_UNBLOCK_URI, src.accountUnblockUri);
+        jsonObject.addProperty(MEMBER_PAYEE_UID, src.payeeUid);
+        jsonObject.addProperty(MEMBER_HOLD_FOR_PICKUP_LINK, src.holdForPickupLink);
+        if (src.digitalGoods != null) {
+            jsonObject.add(MEMBER_DIGITAL_GOODS, DigitalGoodsTypeAdapter.getInstance().toJsonTree(
+                    src.digitalGoods));
+        }
+        BaseProcessPaymentTypeAdapter.Delegate.serialize(jsonObject, src);
+        return jsonObject;
+    }
+
+    @Override
+    protected Class<ProcessPayment> getType() {
+        return ProcessPayment.class;
+    }
+
+    private static DigitalGoods deserializeDigitalGoods(JsonObject jsonObject) {
+        if (jsonObject.has(MEMBER_DIGITAL_GOODS)) {
```

Comment:
I think it is better for performance to get the member first, check if it is not null and perform deserialization. Also safer.